File: source/Expenses.SQLLinux/MyExpenses.Web/server/model/Catalog/Product.js
'use strict';

module.exports = function (sequelize, DataTypes) {
    return sequelize.define('Product', {
        id: {
            type: DataTypes.INTEGER,
            allowNull: false,
            primaryKey: true
        },
        title: {
            type: DataTypes.STRING,
            allowNull: false
        },
        description: {
            type: DataTypes.STRING,
            allowNull: false
        },
        price: {
            type: DataTypes.FLOAT,
            allowNull: false
        },
        creationDate: {
            type: DataTypes.DATE,
            allowNull: false
        },
        available: {
            type: 'BIT',
            allowNull: false
        },
        productCategoryId: {
            type: DataTypes.INTEGER,
            allowNull: false,
            references: {
                model: 'ProductCategory',
                key: 'Id'
            }
        },
        largePicture: {
            type: 'VARBINARY',
            allowNull: true
        },
        thumbnailPicture: {
            type: 'VARBINARY',
            allowNull: true
        },
        //additionalInformation: {
        //    type: DataTypes.STRING,
        //    allowNull: true
        //}        
    }, {
        tableName: 'Product',
        schema: 'Catalog',
        timestamps: false

    });
};

File: source/MyExpenses/MyExpensesDataGenerator/DataModel/Product.cs
namespace MyExpensesDataGenerator
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("Catalog.Product")]
    public partial class Product
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Product()
        {
            PurchaseOrderItems = new HashSet<PurchaseOrderItem>();
        }

        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string Title { get; set; }

        [Required]
        [StringLength(500)]
        public string Description { get; set; }

        public double Price { get; set; }

        [Column(TypeName = "datetime2")]
        public DateTime CreationDate { get; set; }

        public bool Available { get; set; }

        public short ProductCategoryId { get; set; }

        public byte[] LargePicture { get; set; }

        public byte[] ThumbnailPicture { get; set; }

        public virtual ProductCategory ProductCategory { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<PurchaseOrderItem> PurchaseOrderItems { get; set; }
    }
}
